components/support/error/tests/returns_result_but_incorrect_error.stderr

```diff
@@ -0,0 +1,11 @@
+error[E0308]: mismatched types
+  --> tests/returns_result_but_incorrect_error.rs:50:2
+   |
+50 |  #[handle_error]
+   |  ^^^^^^^^^^^^^^^ expected struct `ExternalError`, found struct `OtherExternalError`
+51 |  fn func() -> Result<String, ExternalError> {
+   |               ----------------------------- expected `std::result::Result<String, ExternalError>` because of return type
```

Comment:
This seems like the most important check on the return type.  How is it implemented?  Is there a way to get a bit more text, like `expected struct ExternalError, because body returns a type that implements GetErrorHandling<E = ExternalError>, found struct OtherExternalError instead`

Reply:
It's delegated to the Rust compiler (i.e It's not implemented by the macro). We can try to be smart about it and implement it, but we'd need the macro to "know" the body's return type (as an argument for example). But even then it'll be complicated for us at the proc macro level to understand if the type the body returns implements `GetErrorHandling`. 

For now, leaving it to the compiler and having those tests verifying the errors are more or less understandable is a good start